src/platform/src/test/java/org/geoserver/platform/GeoServerResourceLoaderTest.java

```diff
@@ -162,4 +166,22 @@ public void testSetBaseDirectory() throws IOException {
         assertEquals(tempDir, loader.getBaseDirectory());
         assertEquals(mockStore, loader.getResourceStore());
     }
+
+    @Test
+    public void fromRelativeURLTest() throws IOException {
+        GeoServerResourceLoader loader = new GeoServerResourceLoader();
+        tempFolder.create();
+        File tempDir = tempFolder.getRoot();
+        loader.setBaseDirectory(tempDir);
+
+        Resource res = loader.fromURL("file:relative/with/special+characters%23");
+        assertEquals("relative/with/special characters#", res.path());
```

Comment:
I always get frustrated with special characters.

@NielsCharlier can you add a test for the unicode precomposed characters which are known to cause problems on macOS.

> For example, the character "ü" can be encoded as the single code point U+00FC "ü" or as the base character U+0075 "u" followed by the non-spacing character U+0308 "¨"

Reply:
DOne.

Reply:
Thanks, now to check if the macOS build passes :)